File: Queue implementation using an array.cpp
#include <bits/stdc++.h>
#define max 50
using namespace std;
class Queue{

    int front_end=-1;
    int rear_end=-1;
    int size_of_queue=4;
    int queue[max];

 public: void EnqueueOperation(int value)
	{
        cout<<"\nEnqueue Operation: ";
		if (rear_end == size_of_queue-1)
        {
			printf(" queue is full(Insertion Not Possible)");
		}
		else if(front_end==-1  && rear_end==-1)
		{
            rear_end+=1;
			front_end+=1;
			queue[rear_end] = value;
		}
		else
        {
            rear_end++;
            queue[rear_end] = value;
        }
	}
	 void DequeueOperation()
	{
        int delete_data;
        cout<<"\nDequeue Operation: ";
		if (front_end==-1 && rear_end==-1)
        {
			printf(" queue is empty(Deletion Not Possible)");
		}
		else if(front_end == rear_end)
        {
                delete_data=queue[front_end];
                front_end=-1;
                rear_end=-1;
        }
        else
        {
            delete_data=queue[front_end];
			front_end++;

        }
	}
	 void EnqueueDisplay()
	{
		int x;
		cout<<"\nEnqueue Display: ";
		if (front_end == -1 && rear_end==-1)
        {
			cout<<"queue is empty";
		}
		else
        {
            for (x = front_end; x<=rear_end; x++)
            {
                cout<<queue[x]<<"<--";
            }
        }
	}
     void DequeueDisplay()
	{
		int x;
		cout<<"\nDequeue Display: ";
		if (front_end==-1 && rear_end==-1)
        {
			cout<<"queue is empty";
		}
		else
        {
            for (x = front_end; x <=rear_end; x++)
            {
                cout<<"-->"<<queue[x];
            }
        }
	}
	 void position_of_Front_and_Rear()
	{
		if (front_end==-1&&rear_end==-1)
        {
			cout<<" queue is empty";
		}
		else
        {
            printf("\nPosition of Front is: %d", queue[front_end]);
            printf("\nPosition of Rear is: %d", queue[rear_end]);
        }
	}
};
int main(void)
{
    Queue ob;
	ob.EnqueueDisplay();
    cout<<endl;
    ob.EnqueueOperation(20);
	ob.EnqueueDisplay();
    cout<<endl;
	ob.EnqueueOperation(30);
	ob.EnqueueDisplay();
    cout<<endl;
    ob.EnqueueOperation(40);
	ob.EnqueueDisplay();
    cout<<endl;
	ob.EnqueueOperation(50);
	ob.EnqueueDisplay();
    cout<<endl;
	ob.EnqueueOperation(60);
    ob.EnqueueDisplay();
    cout<<endl;
	ob.EnqueueOperation(70);
    ob.EnqueueDisplay();
    cout<<endl;
	ob.EnqueueOperation(80);
	ob.EnqueueDisplay();
    cout<<endl;
	ob.DequeueOperation();
	ob.DequeueDisplay();
    cout<<endl;
	ob.DequeueOperation();
	ob.DequeueDisplay();
    cout<<endl;
    ob.DequeueOperation();
	ob.DequeueDisplay();
    cout<<endl;
    ob.DequeueOperation();
	ob.DequeueDisplay();
    cout<<endl;
    ob.DequeueOperation();
	ob.DequeueDisplay();
    cout<<endl;
	ob.DequeueOperation();
	ob.DequeueDisplay();
    cout<<endl;
    cout<<"\nPosition of Front and Rear in queue: ";
	ob.position_of_Front_and_Rear();
	cout<<endl;
	return 0;
}

File: Queue implementation using an array.py
front_end=-1
rear_end=-1
size_of_queue=4
queue=[None]*size_of_queue

def EnqueueOperation(value):
    global front_end,rear_end,size_of_queue
    print("Enqueue Operation: ")
    if rear_end == size_of_queue-1:
        print("queue is full(Insertion Not Possible)")
    elif front_end ==-1 and rear_end==-1:
        rear_end+=1
        front_end+=1
        queue[rear_end] = value
    else:
        rear_end+=1
        queue[rear_end] = value

def EnqueueDisplay():
    global front_end,rear_end
    if(front_end == -1 and rear_end==-1):
        print("Enqueue Display: queue is Empty")
    else:
        x=rear_end
        for i in range(front_end,x+1):
            print(queue[i],"<--")
            
def DequeueOperation():
    global front_end,rear_end
    print("Dequeue Operation: ")
    if (front_end==-1 and rear_end==-1):
        print("queue is empty(Deletion Not Possible)")
    elif(front_end == rear_end):
        delete_data=queue[front_end]
        front_end= -1
        rear_end= -1
    else:
        delete_data=queue[front_end]
        front_end+=1

def DequeueDisplay():
    global front_end,rear_end
    print("Dequeue Display: ");
    if(front_end==-1 and rear_end==-1):
        print("queue is Empty");
    else:
        x=rear_end
        for i in range(front_end,x+1):
            print("-->",queue[i])

def position_of_Front_and_Rear():
    if (front_end==-1 and rear_end==-1):
        print("queue is Empty")
    else:
        print("Position of Front is: ",queue[front_end])
        print("Position of Rear is:  ",queue[rear_end])
        
EnqueueDisplay()
EnqueueOperation(20)
EnqueueDisplay()
EnqueueOperation(30)
EnqueueDisplay()
EnqueueOperation(40)
EnqueueDisplay()
EnqueueOperation(50)
EnqueueDisplay()
EnqueueOperation(60)
EnqueueDisplay()
EnqueueOperation(70)
EnqueueDisplay()
DequeueOperation()
DequeueDisplay()
DequeueOperation()
DequeueDisplay()
DequeueOperation()
DequeueDisplay()
DequeueOperation()
DequeueDisplay()
DequeueOperation()
DequeueDisplay()
print("Position of Front and Rear in queue: ")
position_of_Front_and_Rear()

